code/controllers/subsystem/lighting.dm

```diff
@@ -36,8 +36,10 @@ var/list/lighting_update_objects  = list() // List of lighting objects queued fo
 	..()
 
 /datum/controller/subsystem/lighting/fire(resumed, init_tick_checks)
-	var/real_tick_limit = CURRENT_TICKLIMIT
-	CURRENT_TICKLIMIT = (real_tick_limit - world.tick_usage)/3
+	var/real_tick_limit
+	if(!init_tick_checks)
+		real_tick_limit = CURRENT_TICKLIMIT
+		CURRENT_TICKLIMIT = (real_tick_limit - world.tick_usage)/3
```

Comment:
So the actual bug is mostly on this line

This line should actually be: `((real_tick_limit - world.tick_usage)/3)+world.tick_usage`

This bug exists to a lesser extent on normal fires to.

Could you maybe fix that while your in here?